lib/wlib/stl/Bitset.h

```diff
@@ -17,149 +17,215 @@
-				number += bitVal * powerVal;
-			}
-
-			return number;
+		uint64_t to_uint64_t() const {
+            if (nBits <= 32) {
+                return to_uint32_t();
+            }
+            return (((uint64_t) m_array[1]) << INT_SIZE) | ((uint32_t) m_array[0]);
 		}
 
 		/**
 		 * Converts the bits into 32 bit unsigned integer
 		 *
 		 * @return unsigned 32 bit integer
 		 */
-		uint32_t to_uint32_t() {
-			uint32_t number = 0;
-
-			for (uint8_t i = 0; i < 32; ++i) {
-				uint32_t powerVal = (uint32_t) ceil(pow(2, i));
-				uint8_t bitVal = (uint8_t) test(i);
-				number += bitVal * powerVal;
-			}
-
-			return number;
+		uint32_t to_uint32_t() const {
+            return (uint32_t) m_array[0];
 		}
 
 		/**
 		 * Converts the bits into 16 bit unsigned integer
 		 *
 		 * @return unsigned 16 bit integer
 		 */
-		uint16_t to_uint16_t() {
-			uint16_t number = 0;
-
-			for (uint8_t i = 0; i < 16; ++i) {
-				uint16_t powerVal = (uint16_t) ceil(pow(2, i));
-				uint8_t bitVal = (uint8_t) test(i);
-				number += bitVal * powerVal;
-			}
-
-			return number;
+		uint16_t to_uint16_t() const {
+			return (uint16_t) (m_array[0] & pow_mask<16>::value);
 		}
 
 		/**
 		 * Converts the bits into 8 bit unsigned integer
 		 *
 		 * @return unsigned 8 bit integer
 		 */
-		uint8_t to_uint8_t() {
-			uint8_t number = 0;
-
-			for (uint8_t i = 0; i < 8; ++i) {
-				uint8_t powerVal = (uint8_t) ceil(pow(2, i));
-				uint8_t bitVal = (uint8_t) test(i);
-				number += bitVal * powerVal;
-			}
-
-			return number;
+		uint8_t to_uint8_t() const {
+			return (uint8_t) (m_array[0] & pow_mask<8>::value);
 		}
 
+        /**
+         * Access operator returns the bit at the given position.
+         * @param i the position of the bit to test
+         * @return the value of the bit
+         */
+        bool operator[](const uint16_t i) const {
+            return test(i);
+        }
+
+        /**
+         * Assignment operator copies the contents of the bitset.
+         * @param b Bitset to assign
+         */
+        Bitset<nBits>& operator=(Bitset<nBits>& b) {
+            int end = ceil_bits<nBits>::value;
+            for (uint16_t i = 0; i < end; i++) {
+                m_array[i] = (m_array[i] & 0) | b.m_array[i];
+            }
+            return *this;
+        }
+
+        /**
+         * Assignment operator copies the contents of the bitset.
+         * @param b Bitset to assign
+         */
+        Bitset<nBits>& operator=(const Bitset<nBits>& b) {
+            int end = ceil_bits<nBits>::value;
+            for (uint16_t i = 0; i < end; i++) {
+                m_array[i] = (m_array[i] & 0) | b.m_array[i];
+            }
+            return *this;
+        }
+
 	private:
-		int m_array[bits/ NO_OF_BITS];
+        /**
+         * Backing array of integers that contain the bites.
+         * Integer type arrays generally have the fastest access
+         * times in C++.
+         */
+		int m_array[ceil_bits<nBits>::value];

```

Comment:
you should use uint32_t instead of int because int size is system dependent and I believe for ``to_uint32`` you want it to be 32 bits

Reply:
true